cirq-core/cirq/circuits/moment.py

```diff
@@ -80,17 +81,26 @@ class Moment:
             are no such operations, returns an empty Moment.
     """
 
-    def __init__(self, *contents: 'cirq.OP_TREE') -> None:
+    def __init__(self, *contents: 'cirq.OP_TREE', _flatten_contents: bool = True) -> None:
```

Comment:
Should we update the serialization to also include this flag? If not, deserialization would still be inefficient even if the serialized list of operations is flat.